assets/js/components/GoogleChart/index.js

```diff
@@ -220,55 +259,63 @@ export default function GoogleChart( props ) {
 		const chart = chartWrapperRef.current.getChart();
 		const chartLayoutInterface = chart?.getChartLayoutInterface();
 		const chartArea = chartLayoutInterface?.getChartAreaBoundingBox();
-		const dataTable = chartWrapperRef.current?.getDataTable();
+		const dataTable = chartWrapperRef.current.getDataTable();
 
 		if ( ! chartLayoutInterface || ! chartArea || ! dataTable ) {
 			return;
 		}
 
 		// Add the dotted line and tooltip for each date marker.
 		dateMarkers.forEach( ( dateMarker, index ) => {
-			const dateFromMarker = new Date( Date.parse( dateMarker.date ) );
+			const dateFromMarker = new Date( dateMarker.date );
+
+			// Don't render this marker if the date is not within the current
+			// date range.
+			if ( ! isDateWithinRange( dateFromMarker ) ) {
+				return;
+			}
 
-			const chartLine = document.querySelector(
-				`#googlesitekit-chart__date-marker-line--${ instanceID }-${ index }`
+			const chartLine = document.getElementById(
+				`googlesitekit-chart__date-marker-line--${ instanceID }-${ index }`
 			);
 			invariant(
 				chartLine,
 				`#googlesitekit-chart__date-marker-line--${ instanceID }-${ index } is missing from the DOM, but required to render date markers.`
 			);
 
 			// Align the dotted line with the date for this marker.
-			chartLine.style.left =
-				Math.floor(
-					chartLayoutInterface.getXLocation( dateFromMarker )
-				) -
-				1 +
-				'px';
-			chartLine.style.top = Math.floor( chartArea.top ) + 18 + 'px';
-			chartLine.style.height = Math.floor( chartArea.height ) - 18 + 'px';
-			chartLine.style.opacity = 1;
+			Object.assign( chartLine.style, {
+				left: `${
+					Math.floor(
+						chartLayoutInterface.getXLocation( dateFromMarker )
+					) - 1
+				}px`,
+				top: `${ Math.floor( chartArea.top ) + iconSize }px`,
+				height: `${ Math.floor( chartArea.height ) - iconSize }px`,
+				opacity: 1,
+			} );
 
 			// Text is optional, so only modify the DOM elements for the tooltip
 			// text if the property was provided.
 			if ( dateMarker.text ) {
-				const tooltip = document.querySelector(
-					`#googlesitekit-chart__date-marker-tooltip--${ instanceID }-${ index }`
+				const tooltip = document.getElementById(
+					`googlesitekit-chart__date-marker-tooltip--${ instanceID }-${ index }`
 				);
 				invariant(
 					tooltip,
 					`#googlesitekit-chart__date-marker-tooltip--${ instanceID }-${ index } is missing from the DOM, but required to render date marker tooltips.`
 				);
 
 				// Align the tooltip component with the date line.
-				tooltip.style.left =
-					Math.floor(
-						chartLayoutInterface.getXLocation( dateFromMarker )
-					) -
-					9 +
-					'px';
-				tooltip.style.top = Math.floor( chartArea.top ) + 'px';
-				tooltip.style.opacity = 1;
+				Object.assign( tooltip.style, {
+					left: `${
+						Math.floor(
+							chartLayoutInterface.getXLocation( dateFromMarker )
+						) - 9
```

Comment:
Do you think this `9` should be `iconSize / 2` ?

Reply:
Yes! Originally I was messing with other values so left things hardcoded, but that works 🙂